### PR TITLE
[KYUUBI #1839] Minor refactor OperationUtil

### PR DESCRIPTION
**Pull request by link3280**
### _Why are the changes needed?_

This is a sub-task of KPIP-2 #1322.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/result/ResultSetUtil.java [hunk 1]
```diff
@@ -18,41 +18,21 @@
...
 import org.apache.flink.table.api.ResultKind;
 import org.apache.flink.table.catalog.Column;
-import org.apache.flink.table.types.DataType;
-import org.apache.flink.table.types.logical.VarCharType;
 import org.apache.flink.types.Row;
 
-/** Utility class for flink operation. */
-public class OperationUtil {
+/** Utility class for flink result sets. */
+public class ResultSetUtil {
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/result/ResultSetUtil.java:28]: How about rewrite using Scala? Then we can avoid some `.asJava`

Comment (link3280) [reply externals/kyuubi-flink-sql-engine/src/main/java/org/apache/kyuubi/engine/flink/result/ResultSetUtil.java:28]: Agreed.

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.apache.flink.table.api.DataTypes
+import org.apache.flink.table.api.ResultKind
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+/** Util for flink result sets. */
+object ResultSetUtil {
+
+  def stringListToResultSet(strings: List[String], columnName: String): ResultSet = {
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:28]: Please adds the comment of this method comments.

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+package org.apache.kyuubi.engine.flink.result;
+
+import org.apache.flink.table.api.DataTypes
+import org.apache.flink.table.api.ResultKind
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+/** Util for flink result sets. */
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:25]:
```suggestion
/** Utility class for constructing the ResultSet. */
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala (hunk 3 above)
Comment (link3280) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:25]: But this is a scala object, not a class. How about `Utility objetct`?  @SteNicholas

Comment (SteNicholas) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala]: @link3280, I have updated the suggestion.

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.flink.table.api.DataTypes
+import org.apache.flink.table.api.ResultKind
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+/** Utility object for building ResultSet. */
+object ResultSetUtil {
+
+  /**
+   * Build a ResultSet with a column name and a list of String values.
```
Comment (SteNicholas) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:29]: Adds the parameters comments.

**Review by SteNicholas — COMMENTED** [review]
I have left one minor comment for the update.

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+   * Build a simple ResultSet with OK message. Returned when SQL commands are executed successfully.
+   * Noted that a new ResultSet is returned each time, because ResultSet is stateful (with its
+   * cursor).
+   *
+   * @return A simple ResultSet with OK message.
+   */
+  def successResultSet: ResultSet =
+    ResultSet.builder.resultKind(ResultKind.SUCCESS_WITH_CONTENT).columns(Column.physical(
+      "result",
+      DataTypes.STRING)).data(Array[Row](Row.of("OK"))).build
```
Comment (yanghua) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:48]: The previous layout seems more clear：

```
     ResultSet.builder()
        .resultKind(ResultKind.SUCCESS_WITH_CONTENT)
        .columns(Column.physical("result", DataTypes.STRING()))
        .data(new Row[] {Row.of("OK")})
        .build();
```

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala (hunk 4 above)
Comment (link3280) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/result/ResultSetUtil.scala:29]: I think the current comments are fine for an internal utility. But there's no harm in adding more comments, so I'm OK with this.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**